velox/vector/BaseVector.h

```diff
@@ -257,15 +257,12 @@ class BaseVector {
   virtual bool equalValueAt(
       const BaseVector* other,
       vector_size_t index,
-      vector_size_t otherIndex) const {
-    static constexpr CompareFlags kEqualValueAtFlags = {
-        false,
-        false,
-        true /*equalOnly*/,
-        CompareFlags::NullHandlingMode::NoStop /*nullHandlingMode**/};
-    // Will always have value because nullHandlingMode is NoStop.
-    return compare(other, index, otherIndex, kEqualValueAtFlags).value() == 0;
-  }
+      vector_size_t otherIndex,
+      CompareFlags compareFlag = {
```

Comment:
@duanmeng I'm not sure this API change is correct. StopAtNull mode requires return type to be std::optional. Otherwise, it is not possible to distinguish between not-equal and encountered-null-while-checking.

Reply:
@mbasmanova I want to add the compare flag in this API to assist with fixing the nested null compare issue in the `contain` function (similar to `array_sort` but it uses BaseVector::equalValueAt). There already exists a `BaseVector::compare` API with a compare flag and optional return value. I have two proposals,
1. Reuse the `BaseVector::compare` to replace the `BaseVector::equalValueAt` in `ArrayConcatFunction`,
2. Add an extra `BaseVector::equalValueAt` with a compare flag and optional return value.

Option 2 may be more appropriate for it is more semantically clear and we could use it to add `DecodedVector::equalValueAt` too. What do you think?

Reply:
@duanmeng I see. Either way works.

Reply:
Actually, only nullHandlingMode from CompareFlags applies to equalValueAt. Hence, it is not very clear to pass CompareFlags to equalValueAt API. Consider using option (1) or maybe adding nullHandlingMode parameter to equalValueAt.

Reply:
@mbasmanova Got it, I've added an extra equalValueAt using nullHandlingMode as its parameter.